clang/test/Driver/sycl-offload.c

```diff
@@ -891,7 +891,9 @@
 // -std=c++17 override check
 // RUN: %clangxx -### -c -fsycl -std=c++14 -xc++ %s 2>&1 | FileCheck -check-prefix=CHECK-STD-OVR %s
 // RUN: %clang_cl -### -c -fsycl /std:c++14 -TP %s 2>&1 | FileCheck -check-prefix=CHECK-STD-OVR %s
```

Comment:
Does this test the branch with `IsMSVC2015Compatible` equaling `true`, or does the overriding scenario go down another code path? If the latter is correct, should we test the former as well?

Reply:
Regardless of `IsMSVC2015Compatible` settings, `-std=c++17` will be used in the case where no `-std` is set by the user.  The only cases available for `-fsycl` is the user setting their value, or the default `-std=c++17`.  Thought about testing for `NOT -std=c++14` but that should be equivalent to checking for `-std=c++17`